solutions/kafka-long-term-logging/terraform/main.tf

```diff
@@ -0,0 +1,125 @@
+terraform {
+  required_providers {
+    aiven = {
+      source  = "aiven/aiven"
+      version = ">= 2.0.0, < 3.0.0"
+    }
+  }
+}
+
+provider "aiven" {
+  api_token = var.aiven_api_token
+}
+
+data "aiven_project" "proj" {
+  project = var.project
+}
+
+
+resource "aiven_kafka" "kf" {
+  project                 = var.project
+  cloud_name              = var.cloud
+  plan                    = "business-4"
+  service_name            = var.kafka_svc
+  maintenance_window_dow  = "saturday"
+  maintenance_window_time = "10:00:00"
+
+  kafka_user_config {
+    kafka_connect   = true
+    kafka_rest      = true
+    kafka_version   = "3.0"
+    schema_registry = true
+
+    kafka {
+      group_max_session_timeout_ms = 70000
+      log_retention_bytes          = 1000000000
+      auto_create_topics_enable    = true
+    }
+  }
+}
+
+resource "aiven_opensearch" "es" {
+  project                 = var.project
+  cloud_name              = var.cloud
+  plan                    = "hobbyist"
+  service_name            = var.es_svc
+  maintenance_window_dow  = "monday"
+  maintenance_window_time = "10:00:00"
+
+  opensearch_user_config {
+
+    index_patterns {
+      max_index_count   = 14
+      pattern           = "logger-*"
+      sorting_algorithm = "creation_date"
+    }
+  }
+}
+
+
+resource "aiven_kafka_connector" "kf-es-conn" {
+  project      = var.project
+  service_name = aiven_kafka.kf.service_name
+  depends_on = [
+    aiven_opensearch.es,
+    aiven_kafka.kf
+  ]
+  connector_name = "kf-es-conn"
+
+  config = {
+    "name"                                = "kf-es-conn",
+    "connector.class"                     = "io.aiven.connect.elasticsearch.ElasticsearchSinkConnector",
+    "transforms"                          = "routeTS",
+    "topics"                              = "logger",
+    "errors.deadletterqueue.topic.name"   = "es-logger-errors",
+    "connection.url"                      = "https://${aiven_opensearch.es.service_host}:${aiven_opensearch.es.service_port}",
+    "connection.username"                 = "avnadmin",
+    "connection.password"                 = "${aiven_opensearch.es.service_password}",
```

Comment:
Btw if you will use `service_uri` for `"connection.url"` it will contain `service_user` and `service_password` as part of URI so theoretically you could skip `"connection.username"` and `"connection.password"`. 
But this is just a note...